### PR TITLE
XataAPIclient docs

### LINKED ISSUE #31: [NEW] Document XataApiClient
This was part of the docs but was removed during some refactoring.

We should document again under an advanced section.

https://github.com/xataio/client-ts/blob/main/packages/client/src/api/client.ts#L24

https://github.com/xataio/website/blob/f93d12edc3c08a86f3def8fc216ea0b5422d4cab/docs/sdk/typescript.md#api-client

Original issue: https://github.com/xataio/docs/issues/54

### PR DESCRIPTION
**Pull request by joan-ing**
Closes #31

## Summary

- **Readd XataAPIclient docs**  - These were lost during the refactor. Need to be readded

---

### Remaining work

The following work still needs to be completed:

- [ ] <List outstanding tasks here>

### Dependencies

- [ ] <Tasks that must be completed before merging this pull request>

---

### Timing

**Release**:

- [ ] When ready
- [ ] Hold for release | Release after: <mm/dd/yy>

### CODE REVIEW

**Review by SferaDev — CHANGES_REQUESTED** [review]
I think these were rightfully removed. Also the code examples are no longer valid.

File: 040-SDK/010-TypeScript/010-overview.mdx [hunk 1]
```diff
@@ -102,6 +166,84 @@ export const MyComponent: FC<MyComponentProps> = ({ records }) => {
...
+```
+
+Once you have initialized the API client, the operations are organized following the same hiearchy as in the [official documentation](https://docs.xata.io). You have different namespaces for each entity (ie. `workspaces`, `databases`, `tables`, `branches`, `users`, `records`...).
+
+```ts
+import { XataApiClient } from '@xata.io/client';
+
+const api = new XataApiClient();
+
+const { id: workspace } = await api.workspaces.createWorkspace({
```
Comment (eemmiillyy) [inline 040-SDK/010-TypeScript/010-overview.mdx:192]:
```suggestion
 api.workspaces.createWorkspace({
      body: { name: workspaceName }
    })
```

File: 040-SDK/010-TypeScript/010-overview.mdx [hunk 2]
```diff
@@ -102,6 +166,84 @@ export const MyComponent: FC<MyComponentProps> = ({ records }) => {
...
+  data: { name: 'workspace' }
+});
+
+const { databaseName } = await api.database.createDatabase({
```
Comment (eemmiillyy) [inline 040-SDK/010-TypeScript/010-overview.mdx:196]:
```suggestion
await newApi.databases.createDatabase({
      body: {region },
      pathParams: {dbName: "", workspaceId:""}
    })
```

File: 040-SDK/010-TypeScript/010-overview.mdx [hunk 3]
```diff
@@ -102,6 +166,84 @@ export const MyComponent: FC<MyComponentProps> = ({ records }) => {
...
+});
+
+await api.branches.createBranch({
+  workspace,
+  region: 'eu-west-1',
+  database: databaseName,
+  branch: 'branch'
+});
+
+await api.tables.createTable({
```
Comment (eemmiillyy) [inline 040-SDK/010-TypeScript/010-overview.mdx:209]:
```suggestion
 await newApi.table.createTable({
      pathParams: { workspace, region, dbBranchName: `${database}:branch`, tableName: 'table' },
    });
```

**Pull request closed without merging**